pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -76,20 +84,58 @@ private void findSafeLiterals(AbstractApexNode<?> node) {
             int index = literal.jjtGetChildIndex();
             if (index == 0) {
                 if (node instanceof ASTVariableDeclaration) {
-                    addVariable(node);
-                } else {
+                    addVariable((ASTVariableDeclaration) node);
+                } else if (node instanceof ASTBinaryExpression) {
                     ASTVariableDeclaration parent = node.getFirstParentOfType(ASTVariableDeclaration.class);
-                    addVariable(parent);
+                    if (parent != null) {
+                        addVariable(parent);
+                    }
+                    ASTAssignmentExpression assignment = node.getFirstParentOfType(ASTAssignmentExpression.class);
+                    if (assignment != null) {
+                        ASTVariableExpression var = assignment.getFirstChildOfType(ASTVariableExpression.class);
+                        if (var != null) {
+                            addVariable(var);
+                        }
+                    }
 
                 }
             }
+        } else {
+            if (node instanceof ASTField) {
+                try {
+                    final Field f = node.getNode().getClass().getDeclaredField("fieldInfo");
```

Comment:
@sgorbaty just to check I'm getting this right. Apex jorje initializes the field just right (and can therefore be retrieved through reflection), but the getter is hardcoded to return null instead of the field? wow...

In that case, I'd probably add a comment for future self not trying to optimize this without checking jorje is fixed.